test/fit_tests/common/fit_common.py

```diff
@@ -0,0 +1,528 @@
+            print "get_bmc_ips: "
+            print "**** BMC IP node count =", len(BMC_LIST), "****"
+
+    return len(BMC_LIST)
+
+# power on/off all compute nodes in the stack via the BMC
+def power_control_all_nodes(state):
+    if state != "on" and state != "off":
+        print "power_control_all_nodes:  invalid state " + state
+        return
+
+    # Get the list of BMCs that we know about
+    node_count = get_bmc_ips()
+
+    # Send power on/off to all of them
+    for bmc in BMC_LIST:
+        return_code = remote_shell('ipmitool -I lanplus -H ' + bmc['ip'] \
+                                   + ' -U ' + bmc['user'] + ' -P ' \
+                                   + bmc['pw'] + ' -R 1 -N 3 chassis power ' + state)
+        if return_code['exitcode'] != 0:
+            print "Error powering " + state + " node: " + bmc['ip']
+
+    return node_count
+
+def mongo_reset():
+    # clears the Mongo database on ORA to default, returns 0 if successful
+    remote_shell('service onrack-conductor stop')
+    remote_shell('/opt/onrack/bin/monorail stop')
+    remote_shell("mongo pxe --eval 'db.dropDatabase\\\(\\\)'")
+    remote_shell('rm -f /var/lib/dhcp/dhcpd.leases')
+    remote_shell('rm -f /var/log/onrack-conductor-event.log')
+    remote_shell('/opt/onrack/bin/monorail start')
+    if remote_shell('service onrack-conductor start')['exitcode'] > 0:
+        return 1
+    return 0
+
+def appliance_reset():
+    return_code = subprocess.call("ipmitool -I lanplus -H " + ARGS_LIST["bmc"] \
+                                  + " -U root -P 1234567 chassis power reset", shell=True)
+    return return_code
+
+def node_select():
+    # returns a list with valid compute node IDs that match ARGS_LIST["sku"] in 'Name' or 'Model' field
+    # and matches node BMC MAC address in ARGS_LIST["obmmac"] if specified
+    # Otherwise returns list of all IDs that are not 'Unknown' or 'Unmanaged'
+    nodelist = []
+    skuid = "None"
+    # check if user specified a single nodeid to run against
+    # user must know the nodeid and any check for a valid nodeid is skipped
+    if ARGS_LIST["nodeid"] != 'None':
+        nodelist.append(ARGS_LIST["nodeid"])
+        return nodelist
+    else:
+        # Find SKU ID
+        skumap = rackhdapi('/api/2.0/skus')
+        if skumap['status'] != 200:
+            print '**** Unable to retrieve SKU list via API.\n'
+            sys.exit(255)
+        for skuentry in skumap['json']:
+            if str(ARGS_LIST['sku']) in json.dumps(skuentry):
+                skuid = skuentry['id']
+        # Collect node IDs
+        catalog = rackhdapi('/api/2.0/nodes')
+        if skumap['status'] != 200:
+            print '**** Unable to retrieve node list via API.\n'
+            sys.exit(255)
+        # Select node by SKU
+        for nodeentry in catalog['json']:
+            if ARGS_LIST["sku"] == 'all':
+                # Select only managed compute nodes
+                if nodeentry['type'] == 'compute':
```

Comment:
Since only compute nodes are selected if ARG `skus` equals to `all`, whether that means if it is not `all`, it's an exception if that node is not compute?